kolibri/plugins/coach/assets/src/modules/pluginModule.js

```diff
@@ -60,13 +60,17 @@ export default {
   },
   actions: {
     setClassList(store, facilityId) {
-      return ClassroomResource.fetchCollection({
-        getParams: { parent: facilityId || store.getters.currentFacilityId, role: 'coach' },
-      })
-        .then(classrooms => {
-          store.commit('SET_CLASS_LIST', classrooms);
+      if (facilityId) {
+        return ClassroomResource.fetchCollection({
+          getParams: { parent: facilityId, role: 'coach' },
         })
-        .catch(error => store.dispatch('handleApiError', error));
+          .then(classrooms => {
+            store.commit('SET_CLASS_LIST', classrooms);
+          })
+          .catch(error => store.dispatch('handleApiError', error));
+      } else {
```

Comment:
As you are returning in the above `if` statement, the `else` here is not needed!

Alternatively, you could do an 'error early' strategy here and do a `!facilityId` check throw the error, and then just carry on with the unindented function.

Reply:
updated to match your second suggestion!